server/mscalendar/settings_notifications.go

```diff
@@ -0,0 +1,137 @@
+package mscalendar
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/settingspanel"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type notificationSetting struct {
+	title       string
+	description string
+	id          string
+	dependsOn   string
+	getCal      func(string) MSCalendar
+}
+
+func NewNotificationsSetting(getCal func(string) MSCalendar) settingspanel.Setting {
+	os := &notificationSetting{
+		title:       "Receive notifications of new events",
+		description: "Do you want to subscribe to new events and receive a message when they are created?",
+		id:          "notification_setting",
+		dependsOn:   "",
+		getCal:      getCal,
+	}
+
+	return os
+}
+
+func (s *notificationSetting) Set(userID string, value interface{}) error {
+	boolValue := false
+	if value == "true" {
+		boolValue = true
+	}
+
+	cal := s.getCal(userID)
+
+	if boolValue {
+		_, err := cal.LoadMyEventSubscription()
+		if err != nil {
+			_, err := cal.CreateMyEventSubscription()
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	_, err := cal.LoadMyEventSubscription()
+	if err == nil {
+		return cal.DeleteMyEventSubscription()
+	}
+	return nil
+}
+
+func (s *notificationSetting) Get(userID string) (interface{}, error) {
+	cal := s.getCal(userID)
+	_, err := cal.LoadMyEventSubscription()
```

Comment:
So we aren't storing their choice in the settings store, and instead are relying on the stored subscription. I suppose this is good so we only have one source of truth and aren't constantly checking if both values are set (setting and subscriptionID) :+1: